sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -266,6 +271,22 @@ private HttpPipeline addBlobUserAgentModificationPolicy(HttpPipeline pipeline) {
             .build();
     }
 
+    private String modifyUserAgentString(String applicationId, Configuration userAgentConfiguration) {
+        Pattern pattern = Pattern.compile(USER_AGENT_MODIFICATION_REGEX);
```

Comment:
We don't need to recompile the pattern each time build*Client is called. This can be made static final.